rocketpy/Dispersion.py

```diff
@@ -3,82 +3,96 @@
 __license__ = "MIT"
 
 
+import ast
 import math
-import types
+from random import choice
 from time import process_time, time
 
 import matplotlib.pyplot as plt
 import numpy as np
 import simplekml
-from IPython.display import display
 from matplotlib.patches import Ellipse
-from numpy.random import *
 
-from rocketpy.utilities import invertedHaversine
-
-from .AeroSurfaces import EllipticalFins, NoseCone, Tail, TrapezoidalFins
 from .Environment import Environment
 from .Flight import Flight
 from .Function import Function
 from .Motor import SolidMotor
 from .Rocket import Rocket
+from .tools import get_distribution, invertedHaversine
 
+try:
+    from functools import cached_property
+except ImportError:
+    from .tools import cached_property
 
-class Dispersion:
+# TODO: How to save Functions? With pickle? Save just the source?
 
-    """Monte Carlo analysis to predict probability distributions of the rocket's
-    landing point, apogee and other relevant information.
+# TODO: create a method that recreates each flight from inputs_log
```

Comment:
@MateusStano , can you remind me what would be the benefits of doing so?

Reply:
This would allow the user to "reconstruct" any of the flights used in the code. It would be really cool to do and easy to implement. Unfortunately, the problem with saving the trigger function and the drag Function makes it impossible. This would require an implementation of pickle

Reply:
I'd say for now it is not in the scope of v1. We can leave implementing pickle as a future task

Reply:
Agreed! As it is already a TODO in the code, no need for opening a new issue